CMSgov/design-system
# [WNMGDS-3143] Fix `analytics={false}` being passed in `ds-modal-dialog`

## Summary

- Fixes `analytics={false}` being passed in `ds-modal-dialog`.
  - Note that we should probably change `parseBooleanAttr` to behave in a more expected way and audit our existing use of it.
- Applies our standard `onAnalyticsEvent` handler [so our proxy function doesn't override our default analytics function](https://github.com/CMSgov/design-system/pull/3296)

## How to test

1. Turn on analytics logging in Storybook and open the dialog in the `ds-modal-dialog` stories. It should log those events to the Actions tab and the console.
2. Run the new unit tests

## Checklist


### If this is a change to code:
